### ISSUE TITLE
Make editor accessible💡

### ISSUE THREAD

**Issue by Chi-teck** [codex-team/editor.js#1510]
Currently the editor is not accessible for screen readers because:

1. It does not provide accessible names for interative elements.
2. It relies on non-iteractive elements for interaction.

Proposed solution:

1. Add accessible names to interactive elements via `aria-label` attribute or visually hidden text.
2. Add `role="button"` attribute to all interactive elements or (better) use `button` tag instead of `div`.
3. Add `aria-hidden="true"` and `focusable="false"` to SVG icons.

**Comment by Hen2609** [comment]
are there plans to implement a11y features into editor.js?

**Comment by aral** [comment]
Was looking to evaluate it but lack of accessibility is a blocker.

**Comment by gabrr** [comment]
Hi, @aral! 
Would you mind sharing which are the accessibility use cases you have?
Or any tools you are used to use for testing?

**Comment by droud-dev** [comment]
Old issue, but as someone who's currently helping a non-profit set up a CMS (Wordpress is so not an option), I've been looking around for accessible editors (think: entire project built to comply with WCAG 2.2 AA/AAA). The Wagtail's built-in Draftail is based on DraftJS which has been archived/discontinued, so need something else. TinyMCE and CKEditor have unacceptable pricing/licensing models. I was hoping for editor.js, but apparently not.

Use cases:
- Website content creators writing richtext content (ex. blog posts)
- Editors reviewing/updating created content before publishing
- Dynamically rendering created content (ex. header-paragraph-image-paragraph into a nice page)
- Extending the editor with additional content check functionality (e.g. readability score, accessibility checklist, etc)

Note: An editor be usable by editors/creators with limited visibility, motor impairment, and colour blindness at the very least
So, for testing, there's Lighthouse, [axe-core](https://github.com/dequelabs/axe-core), and manual (screen reader, keyboard-only nav, high contrast mode, etc) tests.
